app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
```

Comment:
Nice, I did not notice the unique index. What if the mysql version does not support `utf8mb4`, does it compare the strings case-insensitively as well?